File: JebraAzureFunctions/JebraAzureFunctions/Models/PublishStageEventModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace JebraAzureFunctions.Models
{
    class PublishStageEventModel
    {
        [JsonProperty]
        public int stage_id { get; set; }
        [JsonProperty]
        public int course_id { get; set; }
        [JsonProperty]
        public int origin_user_id { get; set; }
        [JsonProperty]
        public int question_id { get; set; }
        [JsonProperty]
        public int inflicted_hp { get; set; }
        [JsonProperty]
        public bool was_correct { get; set; }
        [JsonProperty]
        public string event_time { get; set; }
    }
}

File: client-app/src/models/PublishStageEventModel.ts
import getAzureFunctions from "getAzureFunctions";

// Interface for Publish Stage Event model sent to the backend
export default interface PublishStageEventModel {
    stage_id: number,
    course_id: number,
    origin_user_id: number,
    question_id: number,
    inflicted_hp: number,
    was_correct: number,
    event_time: string
}

export function publishStageEvent(data: PublishStageEventModel) {
    const url = new URL(getAzureFunctions().PublishStageEvent);

    const requestInfo: RequestInit = {
        method: "PUT",
        body: JSON.stringify(data)
    };

    return fetch(url.toString(), requestInfo);
}